src/bp/core/services/middleware/event-storage.ts

```diff
@@ -0,0 +1,114 @@
+import * as sdk from 'botpress/sdk'
+import { KnexExtension } from 'common/knex'
+import { BotpressConfig } from 'core/config/botpress.config'
+import Database from 'core/database'
+import { EventRepository } from 'core/repositories'
+import { TYPES } from 'core/types'
+import { inject, injectable, tagged } from 'inversify'
+import Knex from 'knex'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+
+import { SessionIdFactory } from '../dialog/session/id-factory'
+
+@injectable()
+export class EventStorage {
+  private readonly BATCH_SIZE = 10
+  private readonly TABLE_NAME = 'events'
+  private knex!: Knex & KnexExtension
+  private intervalRef
+  private currentPromise
+
+  private enabled = false
+  private interval!: number
+  private retentionPeriod!: number
+  private batch: sdk.IO.StoredEvent[] = []
+  private ignoredTypes: string[] = []
+  private ignoredProperties: string[] = []
+
+  constructor(
+    @inject(TYPES.Logger)
+    @tagged('name', 'EventStorage')
+    private logger: sdk.Logger,
+    @inject(TYPES.EventRepository) private eventRepo: EventRepository
+  ) {}
+
+  async initialize(botpressConfig: BotpressConfig, database: Database) {
+    const config = _.get(botpressConfig, 'eventStorage')
+    if (!config || !config.enabled) {
+      return
+    }
+
+    this.knex = database.knex
+    this.interval = ms(config.storageInterval)
+    this.retentionPeriod = ms(config.retentionPeriod)
+    this.ignoredTypes = config.ignoredEventTypes || []
+    this.ignoredProperties = config.ignoredEventProperties || []
+    this.enabled = true
```

Comment:
this seems pretty hardcoded, don't you want to get this form the configs ?

Reply:
I'm already getting the values from the configuration, however i'm doing a quick validation when the class is initialized and storing the "safe" values as instance members, to avoid re-computing them everytime the interval ticks